### ISSUE TITLE
`TestOnePerSecond` in the sampling package fails very rarely.

### ISSUE THREAD

**Issue by shogo82148** [aws/aws-xray-sdk-go#180]
`TestOnePerSecond` in the sampling package fails very rarely.

ref. https://github.com/aws/aws-xray-sdk-go/pull/177/checks?check_run_id=394402981#step:6:54

```
--- FAIL: TestOnePerSecond (2.49s)
	assertions.go:237: 
                        
	Error Trace:	reservoir_test.go:46
		
	Error:		Should be true
		
FAIL
coverage: 83.7% of statements
```

**Comment by bhautikpip** [comment]
Hi @shogo82148,

I was able to repro this one. We are still investigating on the cause of this issue. I am not able to repro this in my local machine but seeing this issue on git tests when tests are failing. It's very inconsistent in nature. We might want to address this in this PR (https://github.com/aws/aws-xray-sdk-go/pull/177).

**Comment by bhautikpip** [comment]
Hi @shogo82148,

were you able to repro this in your local environment ? I am not able to repro this in my local environment to find out what the issue is. Let me know if you are able to repro this in your local.

**Comment by bhautikpip** [comment]
Hi @shogo82148,

The root cause of this issue was we were utilizing current system time because of that For example, let's say we start on the incomplete second like 14:51:25.126302 then it will increase the taken value for this second as well as next second and for this second too (14:51:26.000002). We are measuring for 1.5 seconds overall in this test. That is why we were seeing fluctuation of taken (https://github.com/aws/aws-xray-sdk-go/blob/master/strategy/sampling/reservoir_test.go#L44) value and was the root cause of test case failure. To avoid this timing constraints we have implemented mocked clock because of that we will not be see this fluctuation of time values.

**Comment by bhautikpip** [comment]
Fixed and released in v1.0.0 release so resolving.
